upgrade.md

```diff
@@ -26,55 +27,67 @@ Provided you have suitable udev rules in play, this will already do everything y
 
 ### dfu-util
 
-Download or compile a Black Magic Debug (BMD) firmware.
-As with the [bmputil](#bmputil) section above, the same considerations exist with this method for firmware selection.
+Install [`dfu-util`](http://dfu-util.sourceforge.net/). On macOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
 
-Plug in the Black Magic Probe into your computer and run the following command:
+Plug the Black Magic Probe into your computer and run the following command:
 
 ```bash
 sudo dfu-util -d 1d50:6018,:6017 -s 0x08002000:leave -D blackmagic-native.bin
 ```
 
-You will need to install [`dfu-util`](http://dfu-util.sourceforge.net/) package. On MacOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
+To upgrade non-native hardware see the readme of the different [platforms on GitHub](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms).
 
 ```{note}
 If `dfu-util` fails to switch your BMP into bootloader mode, or you feel like you might have **bricked** your BMP, you can also plug in your BMP while holding down the button. This will force the BMP to stay in the bootloader on power up.
 ```
 
+### stlink-tool
+
+An ST-Link can be used to upgrade the firmware using [stlink-tool](<https://github.com/jeanthom/stlink-tool>).
+
+```bash
+stlink-tools blackmagic.bin
```

Comment:
the tool is `stlink-tool`, there is no s here

Reply:
Good catch!